lib/reply.js

```diff
@@ -132,8 +123,14 @@ function wrapOnSendEnd (err, payload) {
 }
 
 function onSendEnd (reply, payload) {
+  if (payload === undefined) {
+    reply.sent = true
+    reply.res.end()
+    return
```

Comment:
In the RFC there are cases where a content-length header is recommended e.g POST to indicate an empty payload. I shouldn't send the payload here let it falling down. Undefined is stringified to undefined and result in empty payload with content-length 0.

Reply:
It's only _recommended_ to be set for certain request methods (like POST), but it is _required_ to _not_ be set for certain status codes.

I updated the main PR description to included a paragraph about other benefits of this PR. If I didn't send the payload here and let execution continue, the response would have a `Content-Length: 0` header (which violates the spec for some cases) and also a `Content-Type: application/json` header (which is unexpected to me as a developer).